### DISCUSSION TITLE
Extend protocol to return struct that conforms

### DISCUSSION THREAD

**Discussion by joshuakcockrell** [skiptools/skip#143] [Q&A] [answered]
<img width="542" alt="Screenshot 2024-06-11 at 5 04 24 PM" src="https://github.com/skiptools/skip/assets/8979400/309998d3-87f1-42e1-93bd-7bca2444a8c1">

Looks like this pattern doesn't work with Skip. I'm trying to find workarounds right now.

```
Usage

let formatter = FormRowFormatter.email

```

`Expression 'Companion' of type 'EmailFormatter.Companion' cannot be invoked as a function. The function 'invoke()' is not found`

**Comment by aabewhite** [comment]
Thanks for reporting it! Does it work when you just replace Self with EmailFormatter?

```
extension FormRowFormatter where Self == EmailFormatter {
    static var email: EmailFormatter { return EmailFormatter() }
}
```

**Reply by joshuakcockrell** [reply]
That doesn't work, unfortunately. It results in this Kotlin

<img width="493" alt="Screenshot 2024-06-12 at 10 42 35 AM" src="https://github.com/skiptools/skip/assets/8979400/031768dc-5b2b-495c-b67d-35b968a1f4d2">



<img width="741" alt="Screenshot 2024-06-12 at 10 45 13 AM" src="https://github.com/skiptools/skip/assets/8979400/c71ff180-0f24-460e-bcd4-70c02f41625b">

**Reply by joshuakcockrell** [reply]
I can just drop this pattern in the meantime.

Instead of 
`let formatter = FormRowFormatter.email`
I will do
`let formatter = EmailFormatter()`

so it's not a blocker for me.

**Reply by aabewhite** [reply, marked as answer]
I'll look into this.

Did you try `static var email: EmailFormatter { return EmailFormatter() }`?

**Reply by joshuakcockrell** [reply]
@aabewhite Looks like that combo works in Swift & Kotlin!

**Reply by aabewhite** [reply]
Good! I'm almost done with a fix so that Self will work too. I'll probably push it tonight along with several other fixes I'm working on

(I also haven't forgotten about @FocusState, but I need to clear out some backlog from vacation first)

**Reply by joshuakcockrell** [reply]
No rush on `@FocusState`. I've got plenty to migrate over!

**Comment by aabewhite** [comment]
This should now be fixed in skip 0.8.50, and you can use Self as you were before. Use File->Packages->Update to Latest to get the latest transpiler with the fix
